Formula/azure-cli.rb

```diff
@@ -23,7 +22,8 @@ class AzureCli < Formula
   end
 
   depends_on "openssl@1.1"
-  depends_on "python@3.9"
+  # Azure CLI is not compatible with Python 3.9 yet
```

Comment:
It seems to build and test fine, what is the problem exactly?

Reply:
Azure CLI relies on lots of third-party dependencies and some of them have compatible issues with Python 3.9. There are errors when running commands using those dependencies. 

Besides, Azure Devops Pipeline environment currently does not have python 3.9 for testing yet. The test for the formula with python@3.9 would fail in our release pipeline.